REGRESSION(258448@main): [GStreamer] Broke imported/w3c/web-platform-tests/media-capabilities webrtc tests

philn opened the pull request:
#### 12eeaf78a6d495d36bae29a8b9c1b8a36953e9ed
<pre>
REGRESSION(258448@main): [GStreamer] Broke imported/w3c/web-platform-tests/media-capabilities webrtc tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=250315">https://bugs.webkit.org/show_bug.cgi?id=250315</a>

Reviewed by Žan Doberšek and Xabier Rodriguez-Calvar.

Fix WebRTC settings notifications on Page for ports not enabling LibWebRTC. Also rebaseline
media-capabilities tests on GLib ports after recent changes in the GStreamer registry scanner.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/media-capabilities/decodingInfo.webrtc-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/media-capabilities/encodingInfo.webrtc-expected.txt:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::settingsDidChange):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerWebRTCProvider.cpp:
(WebCore::GStreamerWebRTCProvider::initializeVideoEncodingCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeVideoDecodingCapabilities):

Canonical link: <a href="https://commits.webkit.org/258826@main">https://commits.webkit.org/258826@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/59ba51cd5abc98d9873dc870fb3ea94c04edaa7e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103066 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12192 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36086 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112315 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/172518 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/107023 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13211 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3094 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95288 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/110568 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108840 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10145 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93341 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/37769 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91982 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24885 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79502 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5608 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26292 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5772 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/2740 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11771 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45794 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6070 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7523 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

calvaris commented on Source/WebCore/platform/mediastream/gstreamer/GStreamerWebRTCProvider.cpp:
```diff
@@ -99,12 +101,13 @@ void GStreamerWebRTCProvider::initializeAudioDecodingCapabilities()
-
+        if (!isSupportingH265 && codec.mimeType == "video/H265"_s)
+            return true;
         return false;
```
Most stupid nit ever: for coherence with the audio encoding version, you can put a blank like after the last `if` before the `return false`.

philn committed:
REGRESSION(258448@main): [GStreamer] Broke imported/w3c/web-platform-tests/media-capabilities webrtc tests
https://bugs.webkit.org/show_bug.cgi?id=250315

Reviewed by Žan Doberšek and Xabier Rodriguez-Calvar.

Fix WebRTC settings notifications on Page for ports not enabling LibWebRTC. Also rebaseline
media-capabilities tests on GLib ports after recent changes in the GStreamer registry scanner.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/media-capabilities/decodingInfo.webrtc-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/media-capabilities/encodingInfo.webrtc-expected.txt:
* Source/WebCore/page/Page.cpp:
(WebCore::Page::settingsDidChange):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerWebRTCProvider.cpp:
(WebCore::GStreamerWebRTCProvider::initializeVideoEncodingCapabilities):
(WebCore::GStreamerWebRTCProvider::initializeVideoDecodingCapabilities):

Canonical link: https://commits.webkit.org/258826@main
